torch/utils/data/dataloader.py

```diff
@@ -134,6 +144,23 @@ def pin_memory_batch(batch):
         return batch
 
 
+def _set_SIGCHLD_handler():
+    global _SIGCHLD_handler_set
+    if _SIGCHLD_handler_set:
+        return
+    previous_handler = signal.getsignal(signal.SIGCHLD)
+
+    def handler(signum, frame):
+        _error_if_any_worker_fails()
+        if callable(previous_handler):
+            previous_handler(signum, frame)
```

Comment:
Can you do this `callable` check only once outside of this function and set it to `None` otherwise? It's not particularly cheap

Reply:
BTW did you verify you can still get exit codes etc. in Python (i.e. we do not interfere with the usual handlers)

Reply:
I just verified and Python can indeed still get the status and exit codes etc.